test/lib/request-with-rate-limit.test.ts

```diff
@@ -0,0 +1,45 @@
+import * as nock from 'nock';
+import Bottleneck from 'bottleneck';
+import { limiterWithRateLimitRetries } from '../../src/lib/request-with-rate-limit';
+import { limiterForScm } from '../../src/lib/limiters';
+
+beforeEach(() => {
+  return nock('https://testUrlOkResponse')
+    .persist()
+    .get(/.*/)
+    .reply(200, ['test 1', 'test 2']);
+});
+
+beforeEach(() => {
+  return nock('https://testUrlRateLimitResponse')
+    .persist()
+    .get(/.*/)
+    .reply(429, 'Rate limit reached');
+});
+
+describe('Testing requestWithRateLimitRetries', () => {
+  const limiter = new Bottleneck({
+    maxConcurrent: 1,
+    minTime: 500,
+  });
+  test('Test requestWithRateLimitRetries iterates returns an array when response code is 200', async () => {
+    const data = await limiterWithRateLimitRetries(
+      'get',
+      'https://testUrlOkResponse',
+      {},
+      limiter,
+      600,
+    );
+    expect(data.body).toHaveLength(2);
+  }, 20000);
+  test('Test requestWithRateLimitRetries iterates 7 times when response code is 429', async () => {
```

Comment:
would be nice to confirm we did try 7 times by spying on https://github.com/snyk-tech-services/snyk-api-import/pull/279/files#diff-51c7d2f88e2710cc7177b815407d82104b536a1274aec6e52da65e9d03c58878R28 and checking it was called 7 times. Not a blocker, but if you have time please add after